### PR TITLE
Tests updated to validate only major version changes

### PR DESCRIPTION
**Pull request by surgupta-msft**
- Updated bundles tests to validate only major version changes in dependencies
- Re-enabled tests run in pipeline
- Taken Test data files from the latest bundles release - [4.12.0](https://azfunc.visualstudio.com/Azure%20Functions/_build/results?buildId=157367&view=results), [Release tag](https://github.com/Azure/azure-functions-extension-bundles/releases/tag/untagged-8746d22b14b2de5ccf75)

### CODE REVIEW

File: ExtensionBundle.Tests/DependencyValidationTests.cs [hunk 1]
```diff
@@ -202,15 +178,26 @@ private class RuntimeFileComparer : IEqualityComparer<RuntimeFile>
         {
             public bool Equals([AllowNull] RuntimeFile x, [AllowNull] RuntimeFile y)
             {
-                return x.AssemblyVersion == y.AssemblyVersion &&
-                    x.FileVersion == y.FileVersion &&
-                    x.Path == y.Path;
+                if (x.AssemblyVersion is null && y.AssemblyVersion is null)
+                {
+                    return true;
+                }
```
Comment (surgupta-msft) [inline ExtensionBundle.Tests/DependencyValidationTests.cs:184]: Adding this check as getting incorrect comparison result when AssemblyVersion is null

File: ExtensionBundle.Tests/TestData/any_any_extensions.deps.json [hunk 2]
```diff
@@ -8,16 +8,16 @@
     ".NETCoreApp,Version=v3.1": {
       "extensions/1.0.0": {
         "dependencies": {
```
Comment (surgupta-msft) [inline ExtensionBundle.Tests/TestData/any_any_extensions.deps.json:10]: Taken these tests files from bundles latest release [4.12.0]( https://azfunc.visualstudio.com/Azure%20Functions/_build/results?buildId=157367&view=results)

File: ExtensionBundle.Tests/DependencyValidationTests.cs (hunk 1 above)
Comment (soninaren) [reply ExtensionBundle.Tests/DependencyValidationTests.cs:184]: The test should fail if Assembly version is null. That is what we are trying to do with TryParse below. If we have a known file that we build that causes this to fail, we can add an exception for that file.

**Review by soninaren — COMMENTED** [review]
just one comment. Looks good otherwise.

File: ExtensionBundle.Tests/DependencyValidationTests.cs (hunk 1 above)
Comment (surgupta-msft) [reply ExtensionBundle.Tests/DependencyValidationTests.cs]: Updated

**Review by soninaren — APPROVED** [review]

**Pull request merged**